google_guest_agent/events/events.go

```diff
@@ -21,12 +21,14 @@ import (
 	"sync"
 
 	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/events/metadata"
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/events/network"
 	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
 )
 
 var (
 	defaultWatchers = []Watcher{
 		metadata.New(),
+		network.NewNetworkWatcher(network.DefaultPipePath),
```

Comment:
We want the watcher registration tied to the/a feature flag.

Reply:
Should this also respect `NetworkInterfaces.Setup` or just tied to the feature flags?

Reply:
I moved this to happen inside the hostname initialization just before subscribing to the events, LMK what you think.